src/main/java/org/dice_research/opal/civet/metrics/ProviderIdentityMetric.java

```diff
@@ -0,0 +1,273 @@
+
+			// Statement StatementWithPublisher = IteratorOverPublisher.nextStatement();
+			// RDFNode Publisher = StatementWithPublisher.getObject();
+			RDFNode Publisher = IteratorOverPublisher.next();
+			// Check if the publisher object is a blank node.
+			// We 1st start with checking if a non-empty blank node.
+			if (Publisher.isAnon()) {
+
+				evaluatePublisher(Publisher, PublisherScore);
+
+			} else if (Publisher.isURIResource()) {
+
+				evaluatePublisher(Publisher, PublisherScore);
+			} else
+				/*
+				 * 1. A Publisher could be an URI of type foaf:org or foaf:name. We checked
+				 * that. 2. A Publisher could be a blanknode of type foaf:org or foaf:name. We
+				 * checked that. 3. A Publisher could be an empty blank node.
+				 */
+				// This is the 3rd case. Award 0 star
+				PublisherScore.put("EmptyBlankNodeForPublisher", 0);
+		}
+
+		/*
+		 * If PublisherInfo=0 then check if dcat:landingPage is available in
+		 * dcat:catalog
+		 */
+		// System.out.println(PublisherScore.size());
+		// System.out.println(PublisherScore.containsKey("EmptyBlankNodeForPublisher"));
+		if (PublisherScore.size() == 0 || PublisherScore.containsKey("EmptyBlankNodeForPublisher")) {
+
+			NodeIterator landingpages = model.listObjectsOfProperty(dataset, DCAT.landingPage);
+
+			while (landingpages.hasNext()) {
+
+				Object landingPage = landingpages.next();
+				// If we do not have any landingPage in store then add it.
+				if (LandingPageStore.size() == 0) {
+					if (isValidURL(landingPage.toString()))
+						LandingPageStore.add(landingPage.toString());
+				}
+
+				else if (LandingPageStore.size() != 0) {
+					// If landing pages are inconsistent then add the landing page to store.
+					if (!(LandingPageStore.contains(landingPage.toString()))) {
+						if (isValidURL(landingPage.toString()))
+							LandingPageStore.add(landingPage.toString());
+					}
+					/*
+					 * At the end if we have only one value in LandingPageStore then we will award 5
+					 * stars to the dataset.
+					 */
+				}
+			}
+
+			if (LandingPageStore.size() == 1)
+				PublisherScore.put("LandingPageScore", 5);
+			else
+				PublisherScore.put("LandingPageScore", 3);
+		
+			// If LandingPage is not there then check for AccessURL in the distribution
+			if (LandingPageStore.size() == 0) {
+
+				NodeIterator DistributionIterator = model.listObjectsOfProperty(dataset,DCAT.distribution);
+				while (DistributionIterator.hasNext()) {
+
+					TotalNumberOfDistributions++;
+
+					Resource Distribution = (Resource) DistributionIterator.next();
+
+					if (Distribution.hasProperty(DCAT.accessURL)) {
+
+						if (isValidURL(Distribution.getProperty(DCAT.accessURL).getObject().toString()))
+							AccessURLStore.add(Distribution.getProperty(DCAT.accessURL).getObject().toString());
+					}
+				}
+
+				// Calculate a score based on availability of AccessURL and store a score in
+				// PublisherScore.
+				if (AccessURLStore.size() > 0) {
```

Comment:
does checking the access urls make sense for publisher identity?
at least the percentace makes no sense.